src/cgame/cg_ents.cpp

```diff
@@ -1353,6 +1379,20 @@ void CG_AddPacketEntities()
 		cent->valid = false;
 	}
 
+	if ( cg_debugPVS.Get() )
+	{
+		if ( highest > cg_highestActiveEntity )
```

Comment:
Why would someone care about what the highest entity number known to the cgame is? Doesn't seem very relevant.

Reply:
> Why would someone care about what the highest entity number known to the cgame is? Doesn't seem very relevant.

Hu… It's not what that code does… 🤨️ It's a debug print to debug the ballooning mechanism. The important part of the debug log is “`went up`” and “`went down`”.

This debug print is hidden behind two closed doors, you need to _both_ enable debug logs and enable `cg_debugPVS` to get this debug print.

In that file there are already existing very verbose debug prints that print a debug message every time an entity enters or leave PVS, and no one complained about it, ever:

```c++
static void CG_CEntityPVSEnter( centity_t *cent )
{
	entityState_t &es = cent->currentState;

	if ( cg_debugPVS.Get() )
	{
		Log::Debug( "Entity %d entered PVS", cent->currentState.number );
	}
```

```c++
static void CG_CEntityPVSLeave( centity_t *cent )
{
	int           i;
	entityState_t *es = &cent->currentState;

	if ( cg_debugPVS.Get() )
	{
		Log::Debug( "Entity %d left PVS", cent->currentState.number );
	}
```

Reply:
> In that file there are already existing very verbose debug prints that print a debug message every time an entity enters or leave PVS, and no one complained about them, ever (but they were very useful to me so I'm glad they were there)

I don't think removing debug messages is a good idea, if they don't bring a big performance cost. If they do though, maybe we can think of a solution to only have that cost on debug builds?

Reply:
Yeah, same. I don't believe this is blocking.